l2l/optimizers/kalmanfilter/optimizer.py

```diff
@@ -0,0 +1,267 @@
+        self.eval_pop.clear()
+
+        individuals = traj.individuals[traj.generation]
+        gamma = traj.gamma
+        ens_res = []
+        ens_fitnesses = []
+
+        # go over all individuals
+        for i in individuals:
+            # optimization
+            ens = np.array(i.ens)
+            ensemble_size = ens.shape[0]
+            # get the score/fitness of the individual
+            fitness_per_individual = traj.current_results[i.ind_idx][1][
+                'loss']
+            ens_fitnesses.append(fitness_per_individual)
+            model_output = traj.current_results[i.ind_idx][1]['out']
+            enkf = EnKF(maxit=traj.maxit,
+                        online=traj.online,
+                        n_batches=traj.n_batches)
+            enkf.fit(ensemble=ens,
+                     ensemble_size=ensemble_size,
+                     u_exact=None,
+                     observations=self.targets,
+                     model_output=model_output,
+                     noise=traj.noise, p=None, gamma=gamma)
+            ens_res.append(enkf.ensemble)
+
+        generation_name = 'generation_{}'.format(traj.generation)
+        traj.results.generation_params.f_add_result_group(generation_name)
+        ens_fitnesses = np.array(ens_fitnesses)
+
+        generation_result_dict = {
+            'generation': traj.generation,
+            'ensemble_fitnesses': ens_fitnesses,
+        }
+        traj.results.generation_params.f_add_result(
+            generation_name + '.algorithm_params', generation_result_dict)
+
+        if traj.generation > 1 and traj.generation % traj.sampling_generation == 0:
+            params, self.best_fitness, self.best_individual = self._new_individuals(
+                traj, ens_fitnesses, individuals)
+            self.eval_pop = [dict(ens=params[i],
+                                  inputs=self.inputs,
+                                  targets=self.targets)
+                             for i in range(traj.pop_size)]
+        else:
+            self.eval_pop = [dict(ens=ens_res[i],
+                                  inputs=self.inputs,
+                                  targets=self.targets
+                                  )
+                             for i in range(traj.pop_size)]
+        traj.generation += 1
+        self._expand_trajectory(traj)
+
+    def _new_individuals(self, traj, fitnesses, individuals):
+        """
+        Sample new individuals by first ranking and then sampling from a
+        gaussian distribution. The
+        """
+        ranking_idx = list(reversed(np.argsort(fitnesses)))
+        best_fitness = fitnesses[ranking_idx][0]
+        best_ranking_idx = ranking_idx[0]
+        best_individual = individuals[best_ranking_idx]
+        # do the decay
+        eps = self.epsilon * np.exp(-self.decay_rate * traj.generation)
+        # now do the sampling
+        params = [
+            self.optimizee_create_new_individuals(self.random_state,
```

Comment:
Doesn't this function signature for `optimizee_create_new_individuals` break the interface defined in `l2l.optimizees.optimizee.py`?

Reply:
I will adapt it.